custom-tests/Makefile

```diff
@@ -1,16 +1,45 @@
-build:
-	docker build -t rapid-debug .
+# Golang EOL overview: https://endoflife.date/go
+DOCKER_GOLANG_IMAGE ?= golang:1.18.2
 
-run:
-	make build
-	docker run -d -p 9000:8080 --cap-add=SYS_PTRACE rapid-debug:latest
+# On ARM hosts, use: make ARCH=arm64 build-init
+# Check host architecture: uname -m
+# x86_64 or arm64
+ARCH ?= x86_64
 
-clean:
-	rm -f aws-lambda-rie
-	docker rm -f rapid-debug
+ifeq ($(ARCH), arm64)
+ GOARCH=arm64
+else
+ GOARCH=amd64
+endif
 
-debug-run: clean build
-	cd .. && make compile-lambda-linux
-	mv ../bin/aws-lambda-rie-x86_64 ./aws-lambda-rie
-	docker run --cap-add=SYS_PTRACE -d -v $(shell pwd)/dlv:/dlv -v $(shell pwd)/lambda-entrypoint.sh:/lambda-entrypoint.sh -v $(shell pwd)/aws-lambda-rie:/usr/local/bin/aws-lambda-rie -p 9000:8080 -p 9001:8081 -p 40000:40000 -p 9563:9563 --env LOCALSTACK_RIE_DEBUG_PORT=40000 --name rapid-debug rapid-debug:latest
+# Limitation: Debugging x86_64 lambdas does not work on ARM machines due to missing ptrace implementation in qemu used by Docker
+# The function aborts with "could not launch process: fork/exec /var/rapid/init: function not implemented"
+# * Discussion: https://github.com/aws/aws-sam-cli/discussions/4706
+# * Docker for Mac: https://github.com/docker/for-mac/issues/5191#issuecomment-834154431
 
+
+all: build
+
+# build & "package" necessary files for debugging
+build build-init: build-rapid build-delve/dlv
+	cp ../bin/aws-lambda-rie-$(ARCH) ./init/var/rapid/init
+	cp ./build-delve/dlv ./init/var/rapid/dlv
+
+build-rapid:
+	cd .. && GC_FLAGS="all=-N -l" make ARCH=$(ARCH) compile-lambda-linux
```

Comment:
We might want to use compile-with-docker for this one too, so we do not depend on the hosts version of go (I have 1.20 installed and it works, still kind of spicey).

Reply:
let's do it